src/main/webapp/projectContainer/SubmissionForm.js

```diff
@@ -224,14 +224,28 @@ const SubmissionForm = hh(class SubmissionForm extends Component {
   };
 
   removeFile = () => {
-    ProjectMigration.removeSubmissionFile(this.state.params.submissionId, this.state.fileToRemove.uuid).then(prev => {
-      const documentsToUpdate = this.state.documents.filter(doc => doc.id !== this.state.fileToRemove.id);
-      this.setState(prev => {
-        prev.documents = documentsToUpdate;
-        return prev;
+    if(!isEmpty(this.state.fileToRemove.uuid)) {
+      ProjectMigration.removeSubmissionFile(this.state.params.submissionId, this.state.fileToRemove.uuid).then(prev => {
+        this.updateDocuments();
+      }).catch(error => {
+        console.error(error);
+        this.setState(prev => {
+          prev.errors.serverError = true;
+          return prev;
+        });
```

Comment:
As a side note, I prefer this style of setting state even if there is no dependency on previous state. Using the same pattern of `setState` is preferable to using many different patterns and leads to less bugs introduced down the road. It's easy to fall into the trap of using the other form ... then someone adds a line that does depend on previous state.